scripts/o3de/o3de/register.py

```diff
@@ -182,17 +182,22 @@ def register_all_engines_in_folder(engines_path: pathlib.Path,
 
 def register_all_projects_in_folder(projects_path: pathlib.Path,
                                     remove: bool = False,
-                                    engine_path: pathlib.Path = None) -> int:
-    return register_all_o3de_objects_of_type_in_folder(projects_path, 'project', remove, False,
-                                                       stop_on_template_folders, engine_path=engine_path)
+                                    engine_path: pathlib.Path = None,
+                                    force: bool = False,
+                                    dry_run: bool = False) -> int:
+    return register_all_o3de_objects_of_type_in_folder(projects_path, 'project', remove, force,
+                                                       stop_on_template_folders, engine_path=engine_path, 
+                                                       dry_run=dry_run)
 
 
 def register_all_gems_in_folder(gems_path: pathlib.Path,
                                 remove: bool = False,
                                 engine_path: pathlib.Path = None,
-                                project_path: pathlib.Path = None) -> int:
-    return register_all_o3de_objects_of_type_in_folder(gems_path, 'gem', remove, False, stop_on_template_folders,
-                                                       engine_path=engine_path)
+                                project_path: pathlib.Path = None,
+                                force: bool = False,
+                                dry_run: bool = False) -> int:
+    return register_all_o3de_objects_of_type_in_folder(gems_path, 'gem', remove, force, stop_on_template_folders,
```

Comment:
Can you update the `register_all_templates_in_folder` and `register_all_restricted_in_folder` functions as well to accept the `force` and `dry-run` parameters.